internal/api/http/server/router.go

```diff
@@ -19,82 +20,586 @@ func (s *Server) Run(addr string) error {
+		if submissionNumberStr == "" {
+			http.Error(w, "submissionNumber is required.", http.StatusBadRequest)
+			return
+		}
+
+		// Convert parameters to integers
+		taskID, err := strconv.Atoi(taskIDStr)
+		if err != nil {
+			http.Error(w, "Invalid taskID.", http.StatusBadRequest)
+			return
+		}
+
+		userID, err := strconv.Atoi(userIDStr)
 		if err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
+			http.Error(w, "Invalid userID.", http.StatusBadRequest)
 			return
 		}
-		defer file.Close()
 
-		// Set the header for the file
-		fileStat, err := file.Stat()
+		submissionNumber, err := strconv.Atoi(submissionNumberStr)
 		if err != nil {
-			http.Error(w, err.Error(), http.StatusInternalServerError)
+			http.Error(w, "Invalid submission number.", http.StatusBadRequest)
 			return
 		}
+
+		// Retrieve the user's submission file content
+		fileContent, fileName, err := ts.GetUserSubmission(taskID, userID, submissionNumber)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to retrieve submission file: %v", err), http.StatusInternalServerError)
+			return
+		}
+
+		// Set response headers to prompt file download with the original file name
 		w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=%s", fileName))
 		w.Header().Set("Content-Type", "application/octet-stream")
-		w.Header().Set("Content-Length", fmt.Sprintf("%d", fileStat.Size()))
+		w.Header().Set("Content-Length", fmt.Sprintf("%d", len(fileContent)))
 
-		// Copy the file data to the response writer
-		io.Copy(w, file)
+		// Write file content to the response
+		if _, err := w.Write(fileContent); err != nil {
+			http.Error(w, "Failed to write file content to response", http.StatusInternalServerError)
+			return
+		}
 	})
 
-	return &Server{mux: mux}
+	mux.HandleFunc("/getInputOutput", func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+			return
+		}
+
+		// Extract parameters
+		taskIDStr := r.URL.Query().Get("taskID")
+		if taskIDStr == "" {
+			http.Error(w, "taskID is required.", http.StatusBadRequest)
+			return
+		}
+
+		inputOutputIDStr := r.URL.Query().Get("inputOutputID")
+		if inputOutputIDStr == "" {
+			http.Error(w, "inputOutputID is required.", http.StatusBadRequest)
+			return
+		}
+
+		// Convert parameters to integers
+		taskID, err := strconv.Atoi(taskIDStr)
+		if err != nil {
+			http.Error(w, "Invalid taskID.", http.StatusBadRequest)
+			return
+		}
+
+		inputOutputID, err := strconv.Atoi(inputOutputIDStr)
+		if err != nil {
+			http.Error(w, "Invalid inputOutputID.", http.StatusBadRequest)
+			return
+		}
+
+		// Call GetTaskFiles to retrieve the task files as a .tar.gz archive
+		tarFilePath, err := ts.GetInputOutput(taskID, inputOutputID)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Failed to retrieve Input and Output files: %v", err), http.StatusInternalServerError)
```

Comment:
Same as with user Submission, if files do not exist, this is more of a Not Found error, rather then internal server error